src/lms-ds-loader/tests/test_lms_filesystem_provider.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import os
+import pytest
+
+from lms_ds_loader.lms_filesystem_provider import LmsFilesystemProvider
+
+
+class Test_LmsFilesystemProvider:
+    class Test_constructor:
+        def test_given_base_path_is_none_then_raise_error(self):
+            with pytest.raises(AssertionError):
+                LmsFilesystemProvider(None)
+
+        def test_given_base_path_is_whitespace_then_raise_error(self):
+            with pytest.raises(AssertionError):
+                LmsFilesystemProvider("    ")
+
+    class Test_when_getting_all_files:
+        def test_given_invalid_file_path_then_raise_error(self, mocker):
+            csv_path = "does-not-exist"
+
+            mocker.patch.object(os.path, "exists", return_value=False)
+
+            with pytest.raises(OSError):
+                LmsFilesystemProvider(csv_path).get_all_files()
+
+        def test_given_valid_path_does_not_contain_user_directory_then_do_nothing(
+            self, fs
+        ):
+            csv_path = "exists"
+
+            # Arrange
+            fs.create_dir(csv_path)

```

Comment:
This is using [pyfakefs](https://pypi.org/project/pyfakefs/) to fully mock out the filesystem.